jupyter-lib/shared-compiler/src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/notebook/impl/NotebookImpl.kt

```diff
@@ -24,20 +24,36 @@ import org.jetbrains.kotlinx.jupyter.api.libraries.Variable
 import org.jetbrains.kotlinx.jupyter.codegen.FieldsProcessorInternal
 import org.jetbrains.kotlinx.jupyter.codegen.ResultsRenderersProcessor
 import org.jetbrains.kotlinx.jupyter.codegen.TextRenderersProcessorWithPreventingRecursion
+import org.jetbrains.kotlinx.jupyter.debug.DEBUG_THREAD_NAME
 import org.jetbrains.kotlinx.jupyter.libraries.parseLibraryDescriptor
 import org.jetbrains.kotlinx.jupyter.repl.EvalData
 import org.jetbrains.kotlinx.jupyter.repl.ReplRuntimeProperties
 import org.jetbrains.kotlinx.jupyter.repl.SharedReplContext
 import org.jetbrains.kotlinx.jupyter.repl.notebook.MutableCodeCell
 import org.jetbrains.kotlinx.jupyter.repl.notebook.MutableNotebook
 import org.jetbrains.kotlinx.jupyter.util.JupyterClientDetector
+import kotlin.concurrent.thread
 
 class NotebookImpl(
     private val runtimeProperties: ReplRuntimeProperties,
     override val commManager: CommManager,
     private val explicitClientType: JupyterClientType?,
     override val libraryLoader: LibraryLoader,
+    debugPortProvided: Boolean,
 ) : MutableNotebook {
+    private fun `$debugMethod`() {
+        try {
+            while (true) {
+                Thread.sleep(1000)
+            }
+        } catch (_: InterruptedException) {
+        }
+    }
+    private val debugThread = thread(
```

Comment:
Why should we create this object if we even don't run it? I suggest to make it nullable

Reply:
Agreed, let's do it